main.go

```diff
@@ -204,21 +238,31 @@ func (r *reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		logger = logger.WithValues("uid", obj.GetUID(), "reconciliation_action", "upsert")
 	}
 
-	ctx = log.IntoContext(ctx, logger)
-	if err := r.store.Upsert(ctx, requestID, obj, r.gvk.PreferredVersion, r.orgID, deleted); err != nil {
-		var httpErr *backend.HTTPError
-		if errors.As(err, &httpErr) {
-			// when zap finds an `error` type in the values, it will call `Error` and print that
-			// message in the logs. We want to get the underlying values though, to have indexed
-			// fields.
-			logger = logger.WithValues("http_response_error", httpErr.Values())
+	var latestError error
+	for _, orgID := range orgs {
+		requestID := uuid.New().String()
+		logger = logger.WithValues("organization_id", orgID, "request_id", requestID)
+		ctx = log.IntoContext(ctx, logger)
+		if err := r.store.Upsert(ctx, requestID, obj, r.gvk.PreferredVersion, orgID, deleted); err != nil {
+			var httpErr *backend.HTTPError
+			if errors.As(err, &httpErr) {
+				// when zap finds an `error` type in the values, it will call `Error` and print that
+				// message in the logs. We want to get the underlying values though, to have indexed
+				// fields.
+				logger = logger.WithValues("http_response_error", httpErr.Values())
+			}
+
+			logger.Error(fmt.Errorf("could not upsert to store: %w", err), "failed reconciliation")
+			latestError = err
```

Comment:
Btw since Go 1.20, `fmt.Errorf` (and `errors.As` / `errors.Is`) support wrapping multiple errors.
With this we could do:
```go
var allErrs error
for _, orgId := range orgs {
    ...
    if err != nil {
        allErs = fmt.Errorf("%w, %w", err, allErs)
    }
}
```

(this does wrap an initial `nil` value as well, which we could fix with a check)

---

The current code swallows all errors and makes the built-in controller-runtime logging not as useful anymore. It would be nice if we could return all errors.